staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec_test.go

```diff
@@ -1058,25 +1063,27 @@ func TestTLSCredentials(t *testing.T) {
 	if err := a.UpdateTransportConfig(tc); err != nil {
 		t.Fatal(err)
 	}
+	rt, err := transport.New(tc)
+	if err != nil {
+		t.Fatal(err)
+	}
+	client := http.Client{
+		Transport: rt,
+	}
 
 	get := func(t *testing.T, desc string, wantErr bool) {
 		t.Run(desc, func(t *testing.T) {
-			tlsCfg, err := transport.TLSConfigFor(tc)
-			if err != nil {
-				t.Fatal("TLSConfigFor:", err)
-			}
-			client := http.Client{
-				Transport: &http.Transport{TLSClientConfig: tlsCfg},
-			}
 			resp, err := client.Get(server.URL)
 			switch {
 			case err != nil && !wantErr:
 				t.Errorf("got client.Get error: %q, want nil", err)
 			case err == nil && wantErr:
 				t.Error("got nil client.Get error, want non-nil")
 			}
-			if err == nil {
-				resp.Body.Close()
+			if resp != nil && resp.Body != nil {
+				// drain and close body to reuse http keep-alive TCP connections
```

Comment:
I think that this is because `NewUnstartedServer` is not set to use http2, should we:
- test only http2 			`server.EnableHTTP2 = true`
- add a new loop to test http1 and http2

Reply:
I don't think I actually need this specific change on this PR (my old attempt needed it).

Reply:
yeah, sorry, I noticed it and I'm worried that we are only testing http1 with this plugin.
I can follow up with http2